tests/lib/test_io.py

```diff
@@ -0,0 +1,203 @@
+# pylint: disable=missing-function-docstring disable=missing-module-docstring
+
+import os
+from pathlib import Path
+import shutil
+import subprocess
+import pytest
+
+import nestor_api.lib.io as io
+from nestor_api.config.config import Configuration
+
+
+def test_copy_single_file():
+    fixtures_directory_path = Path(
+        os.path.dirname(__file__), '..', '__fixtures__', 'example.yaml').resolve()
+    destination_path = Path(
+        os.path.dirname(__file__), '..', '__fixtures__', 'example_copy.yaml').resolve()
+
+    io.copy(fixtures_directory_path, destination_path)
+
+    assert os.path.isfile(destination_path)
+
+    os.remove(destination_path)
+
+
+def test_copy_directory_with_content():
+    fixtures_directory_path = Path(os.path.dirname(__file__), '..', '__fixtures__').resolve()
+    destination_path = Path(os.path.dirname(__file__), '..', '__fixtures__copy').resolve()
+    fixtures_files = os.listdir(fixtures_directory_path)
+
+    io.copy(fixtures_directory_path, destination_path)
+
+    copied_fixtures_files = os.listdir(destination_path)
+
+    assert os.path.isdir(destination_path)
+    assert fixtures_files == copied_fixtures_files
+
+    shutil.rmtree(destination_path)
+
+
+def test_copy_should_raise_if_failure():
+    with pytest.raises(Exception):
+        fixtures_directory_path = Path(os.path.dirname(__file__), '..', '__abcdefg').resolve()
+        destination_path = Path(os.path.dirname(__file__), '..', '__abcdefg_copy').resolve()
+
+        io.copy(fixtures_directory_path, destination_path)
+
+
+def test_create_temporary_directory():
+    temporary_path = io.create_temporary_directory("prefix")
+
+    assert temporary_path.startswith('/tmp/nestor/work/prefix-')
```

Comment:
Wondering if we should not compare to config variable here.
It is nice to have a plain string in tests, but if, for any case, the `NESTOR_WORK_PATH` variable is set, this test will fail, what do you think?

Reply:
Or maybe overrides values for the whole test file

Reply:
I think since we test already the `get_working_path` that retrieves the config underneath, it's not worth re-testing it again through it.